parser/parser.go

```diff
@@ -1075,46 +1077,86 @@ func (p *Parser) parseFormatStringOperator() (token.Token, string, string, error
 	}
 
 	maxTextLength := p.maxLineLength
+	numLines := p.numLines
+	cursorOverlapWidth := p.cursorOverlapWidth
+	paramValue := ""
+
+	namedParameters := map[string]func(){
+		"fontId": func() {
+			fontID = paramValue
+			fontIdToken = p.peekToken
+		},
+		"maxLineLength": func() {
+			num, _ := strconv.ParseInt(paramValue, 0, 64)
+			maxTextLength = int(num)
+		},
+		"numLines": func() {
+			num, _ := strconv.ParseInt(paramValue, 0, 64)
+			numLines = int(num)
+		},
+		"cursorOverlapWidth": func() {
+			num, _ := strconv.ParseInt(paramValue, 0, 64)
+			cursorOverlapWidth = int(num)
+		},
+	}
+
+	for !p.peekTokenIs(token.RPAREN) {
+		if !p.peekTokenIs(token.COMMA) {
+			return token.Token{}, "", "", NewParseError(p.peekToken, fmt.Sprintf("invalid format() parameter '%s'. Expected a comma", p.peekToken.Literal))
+		}
 
-	if p.peekTokenIs(token.COMMA) {
 		p.nextToken()
-		if p.peekTokenIs(token.STRING) {
+
+		if p.peekTokenIs(token.INT) {
+			num, _ := strconv.ParseInt(p.peekToken.Literal, 0, 64)
+			maxTextLength = int(num)
 			p.nextToken()
-			fontID = p.curToken.Literal
-			fontIdToken = p.curToken
-			if p.peekTokenIs(token.COMMA) {
-				p.nextToken()
-				if err := p.expectPeek(token.INT); err != nil {
-					return token.Token{}, "", "", NewParseError(p.peekToken, fmt.Sprintf("invalid format() maxLineLength '%s'. Expected integer", p.peekToken.Literal))
-				}
-				num, _ := strconv.ParseInt(p.curToken.Literal, 0, 64)
-				maxTextLength = int(num)
-			}
-		} else if p.peekTokenIs(token.INT) {
+			continue
+		}
```

Comment:
Unfortunately this logic isn't quite going to work how we want.  For example, this will end up successfully 
parsing the following example and result in `maxTextLength` being `50`:
```
format("stuff", 10, 20, 30, 40, 50)
```
However, we'd want that to result in an error.

I would rework the high-level logic to this:

```
if (peekTokenIs(STRING) || peekTokenIs(INT)) {
    doOldLogic() // You could make a new function for this
} else {
   parseNamedParams()
}
```

Reply:
Heads up, I amended the above comment.

Reply:
https://github.com/huderlem/poryscript/pull/59/commits/66a27422f5c95705194fd1264bf4ce71d19b75bc